VMBackup/main/freezesnapshotter.py

```diff
@@ -251,6 +253,125 @@ def freeze(self):
         
         return run_result, run_status
 
+    def GetBlobProperties(self, blobUri):
+        blobProperties = None
+        if(blobUri is not None):     
+            try:
+                http_util = HttpUtil(self.logger)
+                sasuri_obj = urlparser.urlparse(blobUri+ '&comp=metadata')
+                headers = {}
+
+                result, httpResp, errMsg = http_util.HttpCallGetResponse('GET', sasuri_obj, None, headers = headers)
+                self.logger.log("FS : GetBlobMetadata: HttpCallGetResponse : result :" + str(result) + ", errMsg :" + str(errMsg))
+                blobProperties = self.httpresponse_get_blob_properties(httpResp)
+            except Exception as e:
+                self.logger.log("FS : GetBlobMetadata: Failed to get blob properties with error: %s, stack trace: %s" % (str(e), traceback.format_exc()))
+        return blobProperties
+
+    def GetHeaderSize(self, headers):
+    # max size of blob metadata 
+            return sys.getsizeof(json.dumps(headers))
+
+
+    def httpresponse_get_blob_properties(self, httpResp):
+        blobProperties = {}
+        if(httpResp != None):
+            self.hutil.log("httpresponse_get_blob_properties: Blob-properties response status:"+str(httpResp.status))
+            if(httpResp.status == 200):
+                resp_headers = httpResp.getheaders()
+                blobProperties = resp_headers
+        return blobProperties
+
+    def populate_snapshotreq_headers_perblob(self, sasuri, sasuri_index, backup_meta_data, blobMetadataTelemetryMessage):
```

Comment:
is it specific to snapshot. if not get also move this util and also rename to remove snapshot